store/file/header.go

```diff
@@ -0,0 +1,94 @@
+package file
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/celestiaorg/celestia-node/share"
+)
+
+const headerSize = 64
+
+type header struct {
+	version  fileVersion
+	fileType fileType
+
+	// Taken directly from EDS
+	shareSize  uint16
+	squareSize uint16
+
+	datahash share.DataHash
+}
+
+type fileVersion uint8
+
+const (
+	fileV0 fileVersion = iota
+)
+
+type fileType uint8
+
+const (
+	ods fileType = iota
+	q1q4
+)
+
+func (h *header) WriteTo(w io.Writer) (int64, error) {
+	b := bytes.NewBuffer(make([]byte, 0, headerSize))
+	_ = b.WriteByte(byte(h.version))
+	_ = b.WriteByte(byte(h.fileType))
+	_ = binary.Write(b, binary.LittleEndian, h.shareSize)
+	_ = binary.Write(b, binary.LittleEndian, h.squareSize)
+	_, _ = b.Write(h.datahash)
+	// write padding
+	_, _ = b.Write(make([]byte, headerSize-b.Len()-1))
+	return writeLenEncoded(w, b.Bytes())
+}
+
+func readHeader(r io.Reader) (*header, error) {
+	bytesHeader, err := readLenEncoded(r)
+	if err != nil {
+		return nil, err
+	}
+	if len(bytesHeader) != headerSize-1 {
+		return nil, fmt.Errorf("readHeader: read %d bytes, expected %d", len(bytesHeader), headerSize)
+	}
+	h := &header{
+		version:    fileVersion(bytesHeader[0]),
+		fileType:   fileType(bytesHeader[1]),
+		shareSize:  binary.LittleEndian.Uint16(bytesHeader[2:4]),
+		squareSize: binary.LittleEndian.Uint16(bytesHeader[4:6]),
+		datahash:   make([]byte, 32),
+	}
+
+	copy(h.datahash, bytesHeader[6:6+32])
+	return h, err
+}
+
+func writeLenEncoded(w io.Writer, data []byte) (int64, error) {
+	_, err := w.Write([]byte{byte(len(data))})
```

Comment:
Waaaait, are we writing just the lower 8bits of `len(data)` here?

Reply:
Ok, looks like we always(?) write 64 bytes (`headerSize`), am I right?

Reply:
It writes len of data into first byte. Right now we read 64 bytes always no matter what. In future header struct might grow and new fields might be added. It would require old storages to be upgraded to bigger header or lenEncoded version of it. So lenEncoding is added to prevent future break of full file. Otherwise we would need to rewrite all files fully on file.header upgrade. 
TLDR: It doesn't do much as of now, but will be needed if header struct grow to keep backwards compatibility.

Reply:
Sounds like a good place for a) write `int32` which will be enough for any sane version (also, it's header, I bet it will be 1Gb) or b) specify header version with byte counter.

Anyway, both of such solutions will make simpler our migration in the future (even if it will happen).

Reply:
I've removed length encoding in favour of headerVersion